### PR TITLE
chore: remove `tokio-trace`, add "Related Projects" to README (v0.1)

### PR DESCRIPTION
**Pull request by hawkw**
## Motivation

The `tokio-trace` and `tokio-trace-core` crates have been renamed to
`tracing` and `tracing-core`, and moved to their own repository
(`tokio-rs/tracing`).

## Solution

This branch removes `tokio-trace` and `tokio-trace-core` from the
`tokio` repository. In addition, I've added a "Related Projects" section
to the root README, which lists `tracing` (as well as  `mio`, and
`bytes`) as other libraries maintained by the Tokio project. I thought
that this would help folks looking for `tokio-trace` here find it in its
new home.

In addition, it changes `tokio` to depend on `tracing-core` rather than
`tokio-trace-core`.

This is the same change as #1221, but backported to the v0.1 branch.

Signed-off-by: Eliza Weisman <eliza@buoyant.io>

### CODE REVIEW

**Comment by jonhoo** [comment]
I _think_ you missed the `extern crate` in `tokio/src/lib.rs`?

File: tokio/Cargo.toml [hunk 1]
```diff
@@ -47,7 +47,8 @@ rt-full = [
   "tokio-current-thread",
   "tokio-executor",
   "tokio-threadpool",
-  "tokio-trace-core",
+  "tokio-macros",
```
Comment (jonhoo) [inline tokio/Cargo.toml]: Did you intentionally include `tokio-macros` here?

**Comment by jonhoo** [comment]
Nudge on this. I think it's important to land + release, as otherwise `tracing` basically doesn't work with `tokio` (or rather, it requires you do all the passing of the `Dispatch` yourself).

**Comment by jonhoo** [comment]
Also, CI failure is:
```
cargo test --all
error: failed to parse manifest at `/tmp/cirrus-ci-build/tokio/Cargo.toml`

Caused by:
  Feature `rt-full` includes `tokio-macros` which is neither a dependency nor another feature
```

**Comment by jonhoo** [comment]
I'm trialing fixes to this with passing CI in https://github.com/tokio-rs/tokio/pull/1234 :)

**Comment by hawkw** [comment]
@jonhoo I've merged your changes from #1234 and feature-flagged `tracing-core` as `experimental-tracing` so that we don't pull it in on Rust 1.26 (as @carllerche suggested)

**Comment by jonhoo** [comment]
Yay! How do we go about doing a point release @carllerche ?

**Comment by carllerche** [comment]
@jonhoo open a release PR that does the necessary changes / bumps and I can merge then release :+1:

**Pull request merged**